pkg/cloudprovider/vsphere/instances.go

```diff
@@ -171,5 +171,27 @@ func (i *instances) InstanceExistsByProviderID(ctx context.Context, providerID s
 // InstanceShutdownByProviderID returns true if the instance is in safe state to detach volumes
 func (i *instances) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
 	klog.V(4).Info("instances.InstanceShutdownByProviderID() called")
-	return false, cloudprovider.NotImplemented
+
+	// Check if node has been discovered already
+	uid := GetUUIDFromProviderID(providerID)
+	if _, ok := i.nodeManager.nodeUUIDMap[uid]; !ok {
+		// IF the uuid is not cached, we end up here
+		klog.V(2).Info("instances.InstanceShutdownByProviderID() NOT CACHED")
+		if err := i.nodeManager.DiscoverNode(uid, cm.FindVMByUUID); err != nil {
+			klog.V(4).Info("instances.InstanceShutdownByProviderID() NOT FOUND with ", uid)
+			// if we can't discover, return false with an error in tow
+			return false, err
+		}
+		klog.V(2).Infof("instances.InstanceShutdownByProviderID() EXISTS with %q", uid)
+	}
+
+	vm, err := i.nodeManager.shakeOutNodeIDLookup(ctx, uid, cm.FindVMByUUID)
```

Comment:
We can remove the call to shakeOutNodeIDLookup() because `nodeUUIDMap[uid]` contains `vm            *vclib.VirtualMachine` and that call is fairly expensive.

Reply:
Agree with @dvonthenen . Please use nodeUUIDMap[uid] to retrieve the vm instance.

Reply:
done, I figured there was a lookup table for it, but didn't occur to me it was this obvious 🙂